akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -269,7 +270,7 @@ private[client] object NewHostConnectionPool {
 
                 if (!state.isConnected && connection != null) {
                   debug(s"State change from [${previousState.name}] to [Unconnected]. Closing the existing connection.")
-                  closeConnection()
+                  closeConnection(force = false)
```

Comment:
Should we disambiguate between 'natural' and 'abrupt' closures here?

Reply:
Ah, yes that's pretty interesting. Looking at the SlotStates it seems that there's only a single case where closing would be normal and all the others would be a forced connection closure after a failure.